src/main/java/org/opensearch/sdk/handlers/ExtensionsRestRequestHandler.java

```diff
@@ -77,73 +69,18 @@ public RestExecuteOnExtensionResponse handleRestExecuteOnExtensionRequest(Extens
             );
         }
 
-        // Temporary code to create a RestRequest from the ExtensionRestRequest before header code added
-        // Remove this and replace with SDKRestRequest being generated by this PR:
-        // https://github.com/opensearch-project/opensearch-sdk-java/pull/605
-        RestRequest restRequest = RestRequest.request(sdkNamedXContentRegistry.getRegistry(), new HttpRequest() {
-
-            @Override
-            public Method method() {
-                return request.method();
-            }
-
-            @Override
-            public String uri() {
-                // path strips query off uri but probably want to pass the whole uri
-                // this will make the request behave as expected (without query params)
-                return request.path();
-            }
-
-            @Override
-            public BytesReference content() {
-                return request.content();
-            }
-
-            @Override
-            public Map<String, List<String>> getHeaders() {
-                // This effectively recreates the only header we need right now
-                // PR replacing this will pass more headers
-                XContentType xContentType = request.getXContentType();
-                return xContentType == null ? Collections.emptyMap() : Map.of("Content-Type", List.of(xContentType.mediaType()));
-            }
-
-            @Override
-            public List<String> strictCookies() {
-                return Collections.emptyList();
-            }
-
-            @Override
-            public HttpVersion protocolVersion() {
-                return null;
-            }
-
-            @Override
-            public HttpRequest removeHeader(String header) {
-                // we don't use
-                return null;
-            }
-
-            @Override
-            public HttpResponse createResponse(RestStatus status, BytesReference content) {
-                return null;
-            }
-
-            @Override
-            public Exception getInboundException() {
-                return null;
-            }
-
-            @Override
-            public void release() {}
-
-            @Override
-            public HttpRequest releaseAndCopy() {
-                return null;
-            }
-        }, null);
+        SDKHttpRequest sdkHttpRequest = new SDKHttpRequest(request);
+        SDKRestRequest sdkRestRequest = new SDKRestRequest(
+            sdkNamedXContentRegistry.getRegistry(),
+            request.params(),
+            request.path(),
+            request.headers(),
+            sdkHttpRequest,
```

Comment:
Suggestion: use `new SDKHttpRequest(request)` here, and you won't need the temporary variable on line 72.